### PR TITLE
Added timeout option to choose and other patches (maybe)

### LINKED ISSUE #565: gum choose v0.14.0: timeout no longer working
**Describe the bug**
`gum choose --selected="foo" --timeout=3s "a" "b" "foo" "d"`
timeout no longer working

**To Reproduce**
1. `gum choose --selected="foo" --timeout=3s "a" "b" "foo" "d"`
2. No progress

**Expected behavior**
timeout after 3s

**Desktop (please complete the following information):**
 - OS: macos 14.4.1

Incidentally

https://github.com/charmbracelet/gum/blob/4222e59c25dbd25423285e5d088b8c66d758c87b/choose/options.go#L27

typo for `GUM_CCHOOSE_TIMEOUT`?

### PR DESCRIPTION
**Pull request by csandeep**
* Added timeout option to choose and other patches (maybe)
    -  Fixes #565 

### Changes
- choose/command.go

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -52,3 +53,5 @@ require (
 	golang.org/x/sys v0.21.0 // indirect
 	golang.org/x/text v0.16.0 // indirect
 )
+
+replace github.com/charmbracelet/huh => ../huh
```
Comment (maaslalani) [inline go.mod:57]: I think you need to remove this and bump `huh` here.

Comment (maaslalani) [reply go.mod:57]: You can do that with:

```
go get github.com/charmbracelet/huh@master
```

**Comment by maaslalani** [comment]
LGTM! Tested it out and works well!
Let's add this to the rest of the commands as well!

**Comment by meowgorithm** [comment]
We're reverting this due to a bug where `timeout` is printed on timeout. Additonally:

* this should also default to seconds when no unit is specified
* we should show a timeout indicator in the UI

We'll address these and re-add this after the next release.

**Comment by eread** [comment]
> We're reverting this due to a bug where `timeout` is printed on timeout. Additonally:
> 
> * this should also default to seconds when no unit is specified
> * we should show a timeout indicator in the UI
> 
> We'll address these and re-add this after the next release.

@meowgorithm Any timeframe on when the fix for this will be released?

**Comment by meowgorithm** [comment]
Hey, @eread! By fix, do you mean when will timeout be re-added to `gum choose`? The revision where we reverted the change is in the current release.

**Comment by eread** [comment]
> Hey, @eread! By fix, do you mean when will timeout be re-added to `gum choose`? The revision where we reverted the change is in the current release.

@meowgorithm Yes, that's what I meant. The help text for `gum choose` still mentions `--timeout`, but I don't think it works at all.

Thanks!

**Comment by eread** [comment]
> > Hey, @eread! By fix, do you mean when will timeout be re-added to `gum choose`? The revision where we reverted the change is in the current release.
> 
> @meowgorithm Yes, that's what I meant. The help text for `gum choose` still mentions `--timeout`, but I don't think it works at all.
> 
> Thanks!

@meowgorithm It looks like this PR https://github.com/charmbracelet/gum/pull/632 is designed to restore the `--timeout` functionality?

CC @aymanbagabas

**Comment by eread** [comment]
> > > Hey, @eread! By fix, do you mean when will timeout be re-added to `gum choose`? The revision where we reverted the change is in the current release.
> > 
> > 
> > @meowgorithm Yes, that's what I meant. The help text for `gum choose` still mentions `--timeout`, but I don't think it works at all.
> > Thanks!
> 
> @meowgorithm It looks like this PR #632 is designed to restore the `--timeout` functionality?
> 
> CC @aymanbagabas

CC @maaslalani

**Pull request merged**